client-java/instrumentation-shared/src/main/java/org/evomaster/client/java/instrumentation/shared/dto/ControlDependenceGraphDto.java

```diff
@@ -0,0 +1,200 @@
+package org.evomaster.client.java.instrumentation.shared.dto;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Serializable representation of the control-dependence information of a method.
+ * Contains only the data needed by EvoMaster core: branch objective identifiers,
+ * the subset that are roots, and the parent→child relationships between them.
+ */
+public class ControlDependenceGraphDto implements Serializable {
```

Comment:
if it is under `instrumentation-shared`, then it is not a DTOs. also, here there is plenty of logic that is part of a DTO. a DTO is a POJO, with just public fields.